Java classes generated by `wsdl2java` not passed to Gradle `compileKotlin` since 3.26.1

### Describe the bug

Originally reported as https://github.com/quarkiverse/quarkus-cxf/issues/1962 by @c0ldd

A simple [reproducer project](https://github.com/c0ldd/cxf-codegen-issue) that worked with Platform BOMs 3.26.0 stopped working with Platform BOMs 3.26.1.

Steps to reproduce:

```
got clone git@github.com:c0ldd/cxf-codegen-issue.git
cd cxf-codegen-issue
sed -i 's|3.27.0|3.26.1|g' gradle.properties
./gradlew clean assemble --info --no-daemon
```


### Expected behavior

The project compiles and the classes generated by `wsdl2java` codegen of `quarkus-cxf` are included in the output jar

### Actual behavior

The project fails at compiling Kotlin. 

Observations: 

* The java files were generated by `wsdl2java` codegen of `quarkus-cxf` and can be seen under `build/classes/java/quarkus-generated-sources/wsdl2java`
* Apparently, the `compileKotlin` task is not aware of those and falls on an Unresolved reference 'tempuri'
* The failure is the same for all 3.26.1+ versions and 3.27.0+ versions. The last working version is 3.26.0

Here is the log with 3.26.1:

```
> Task :quarkusGenerateCode
Caching disabled for task ':quarkusGenerateCode' because:
  Build cache is disabled
Task ':quarkusGenerateCode' is not up-to-date because:
  Output property 'generatedOutputDirectory' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources has been removed.
  Output property 'generatedOutputDirectory' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java has been removed.
  Output property 'generatedOutputDirectory' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org has been removed.
  and more...
Starting process 'Gradle Worker Daemon 1'. Working directory: /home/ppalaga/.gradle/workers Command: /home/ppalaga/.sdkman/candidates/java/21.0.8-tem/bin/java -Dplatform.quarkus.native.builder-image=mandrel -Dquarkus.cxf.codegen.wsdl2java.enabled=true -Dquarkus.cxf.codegen.wsdl2java.includes=wsdl_example.wsdl -Dquarkus.cxf.codegen.wsdl2java.validate=true -Dquarkus.cxf.codegen.wsdl2java.wsdl-location=classpath:wsdl_example.wsdl -Dquarkus.home=/home/ppalaga/.sdkman/candidates/quarkus/current @/home/ppalaga/.gradle/.tmp/gradle-worker-classpath6205550466221497228txt -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=IE -Duser.language=en -Duser.variant worker.org.gradle.process.internal.worker.GradleWorkerMain 'Gradle Worker Daemon 1'
Successfully started process 'Gradle Worker Daemon 1'
Started Gradle worker daemon (0.334 secs) with fork options DaemonForkOptions{executable=/home/ppalaga/.sdkman/candidates/java/21.0.8-tem/bin/java, minHeapSize=null, maxHeapSize=null, jvmArgs=[], keepAliveMode=SESSION}.
Generating Quarkus code for org.acme:cxf-codegen-issue:1.0.0-SNAPSHOT
  launch mode:                  NORMAL
  base name:                    cxf-codegen-issue-1.0.0-SNAPSHOT
  generated source directory:   /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources
  build directory:              /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build
Running wsdl2java -d /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java -validate -wsdlLocation classpath:wsdl_example.wsdl /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/resources/wsdl_example.wsdl
Resolve mutations for :checkKotlinGradlePluginConfigurationErrors (Thread[#178,Execution worker,5,main]) started.
:checkKotlinGradlePluginConfigurationErrors (Thread[#180,Execution worker Thread 3,5,main]) started.

> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
Skipping task ':checkKotlinGradlePluginConfigurationErrors' as task onlyIf 'errorDiagnostics are present' is false.
Resolve mutations for :compileKotlin (Thread[#180,Execution worker Thread 3,5,main]) started.
:compileKotlin (Thread[#180,Execution worker Thread 3,5,main]) started.

> Task :compileKotlin FAILED
Caching disabled for task ':compileKotlin' because:
  Build cache is disabled
Task ':compileKotlin' is not up-to-date because:
  Task has failed previously.
The input changes require a full rebuild for incremental task ':compileKotlin'.
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/java', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/grpc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avdl', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avpr', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avsc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/java', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/grpc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avdl', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avpr', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avsc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/java', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/grpc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avdl', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avpr', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avsc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/java', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/grpc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avdl', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avpr', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avsc', not found
Kotlin source files: /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/kotlin/dev/test/SomeCompileClass.kt
Java source files: 
Script source files: 
Script file extensions: 
Using Kotlin/JVM incremental compilation
[KOTLIN] Kotlin compilation 'jdkHome' argument: /home/ppalaga/.sdkman/candidates/java/21.0.8-tem
Options for KOTLIN DAEMON: IncrementalCompilationOptions(super=CompilationOptions(compilerMode=INCREMENTAL_COMPILER, targetPlatform=JVM, reportCategories=[0, 3], reportSeverity=2, requestedCompilationResults=[0], kotlinScriptExtensions=[]), sourceChanges=org.jetbrains.kotlin.buildtools.api.SourcesChanges$Unknown@3812d4a0, classpathChanges=NotAvailableForNonIncrementalRun, workingDir=/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/kotlin/compileKotlin/cacheable, multiModuleICSettings=MultiModuleICSettings(buildHistoryFile=/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/kotlin/compileKotlin/local-state/build-history.bin, useModuleDetection=false), icFeatures=IncrementalCompilationFeatures(usePreciseJavaTracking=true, withAbiSnapshot=false, preciseCompilationResultsBackup=true, keepIncrementalCompilationCachesInMemory=true, enableUnsafeIncrementalCompilationForMultiplatform=false, enableMonotonousIncrementalCompileSetExpansion=true), outputFiles=[/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/kotlin/main, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/kotlin/compileKotlin/cacheable, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/kotlin/compileKotlin/local-state])
e: file:///home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/kotlin/dev/test/SomeCompileClass.kt:11:25 Unresolved reference 'tempuri'.
Finished executing kotlin compiler using DAEMON strategy
Build f1288471-dea9-4bc2-a909-134d0e946710 is closed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details
```


Here is a successful build with 3.26.0:

```
> Task :quarkusGenerateCode
Caching disabled for task ':quarkusGenerateCode' because:
  Build cache is disabled
  Gradle does not know how file 'build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/CalculatorSoap.java' was created (output property 'generatedOutputDirectory'). Task output caching requires exclusive access to output paths to guarantee correctness (i.e. multiple tasks are not allowed to produce output in the same location).
Task ':quarkusGenerateCode' is not up-to-date because:
  Output property 'generatedOutputDirectory' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/Calculator.java has changed.
  Output property 'generatedOutputDirectory' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/CalculatorSoap.java has changed.
Starting process 'Gradle Worker Daemon 1'. Working directory: /home/ppalaga/.gradle/workers Command: /home/ppalaga/.sdkman/candidates/java/21.0.8-tem/bin/java -Dplatform.quarkus.native.builder-image=mandrel -Dquarkus.cxf.codegen.wsdl2java.enabled=true -Dquarkus.cxf.codegen.wsdl2java.includes=wsdl_example.wsdl -Dquarkus.cxf.codegen.wsdl2java.validate=true -Dquarkus.cxf.codegen.wsdl2java.wsdl-location=classpath:wsdl_example.wsdl -Dquarkus.home=/home/ppalaga/.sdkman/candidates/quarkus/current @/home/ppalaga/.gradle/.tmp/gradle-worker-classpath5024466836854262949txt -Xmx512m -Dfile.encoding=UTF-8 -Duser.country=IE -Duser.language=en -Duser.variant worker.org.gradle.process.internal.worker.GradleWorkerMain 'Gradle Worker Daemon 1'
Successfully started process 'Gradle Worker Daemon 1'
Started Gradle worker daemon (0.334 secs) with fork options DaemonForkOptions{executable=/home/ppalaga/.sdkman/candidates/java/21.0.8-tem/bin/java, minHeapSize=null, maxHeapSize=null, jvmArgs=[], keepAliveMode=SESSION}.
Generating Quarkus code for org.acme:cxf-codegen-issue:1.0.0-SNAPSHOT
  launch mode:                  NORMAL
  base name:                    cxf-codegen-issue-1.0.0-SNAPSHOT
  generated source directory:   /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources
  build directory:              /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build
Running wsdl2java -d /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java -validate -wsdlLocation classpath:wsdl_example.wsdl /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/resources/wsdl_example.wsdl
Resolve mutations for :quarkusGenerateDevAppModel (Thread[#178,Execution worker,5,main]) started.
:quarkusGenerateDevAppModel (Thread[#178,Execution worker,5,main]) started.

> Task :quarkusGenerateDevAppModel
Caching disabled for task ':quarkusGenerateDevAppModel' because:
  Build cache is disabled
  Caching has not been enabled for the task
Task ':quarkusGenerateDevAppModel' is not up-to-date because:
  Value of input property 'projectDescriptor' has changed for task ':quarkusGenerateDevAppModel'
Resolve mutations for :quarkusGenerateCodeDev (Thread[#178,Execution worker,5,main]) started.
:quarkusGenerateCodeDev (Thread[#178,Execution worker,5,main]) started.

> Task :quarkusGenerateCodeDev
Caching disabled for task ':quarkusGenerateCodeDev' because:
  Build cache is disabled
  Gradle does not know how file 'build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/CalculatorSoap.java' was created (output property 'generatedOutputDirectory'). Task output caching requires exclusive access to output paths to guarantee correctness (i.e. multiple tasks are not allowed to produce output in the same location).
Task ':quarkusGenerateCodeDev' is not up-to-date because:
  Output property 'generatedOutputDirectory' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/Calculator.java has changed.
  Output property 'generatedOutputDirectory' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/CalculatorSoap.java has changed.
Generating Quarkus code for org.acme:cxf-codegen-issue:1.0.0-SNAPSHOT
  launch mode:                  DEVELOPMENT
  base name:                    cxf-codegen-issue-1.0.0-SNAPSHOT
  generated source directory:   /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources
  build directory:              /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build
Running wsdl2java -d /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java -validate -wsdlLocation classpath:wsdl_example.wsdl /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/resources/wsdl_example.wsdl
Resolve mutations for :checkKotlinGradlePluginConfigurationErrors (Thread[#178,Execution worker,5,main]) started.
:checkKotlinGradlePluginConfigurationErrors (Thread[#190,Execution worker Thread 13,5,main]) started.

> Task :checkKotlinGradlePluginConfigurationErrors SKIPPED
Skipping task ':checkKotlinGradlePluginConfigurationErrors' as task onlyIf 'errorDiagnostics are present' is false.
Resolve mutations for :compileKotlin (Thread[#190,Execution worker Thread 13,5,main]) started.
:compileKotlin (Thread[#190,Execution worker Thread 13,5,main]) started.

> Task :compileKotlin
Caching disabled for task ':compileKotlin' because:
  Build cache is disabled
Task ':compileKotlin' is not up-to-date because:
  Input property 'javaSources' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/Calculator.java has changed.
  Input property 'javaSources' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/CalculatorSoap.java has changed.
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/java', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/grpc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avdl', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avpr', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avsc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/java', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/grpc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avdl', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avpr', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avsc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/java', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/grpc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avdl', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avpr', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avsc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/java', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/grpc', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avdl', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avpr', not found
file or directory '/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/avsc', not found
Kotlin source files: /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/src/main/kotlin/dev/test/SomeCompileClass.kt
Java source files: /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/Multiply.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/Add.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/MultiplyResponse.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/DivideResponse.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/Calculator.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/SubtractResponse.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/AddResponse.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/ObjectFactory.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/CalculatorSoap.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/Subtract.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/Divide.java, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/package-info.java
Script source files: 
Script file extensions: 
Using Kotlin/JVM incremental compilation
[KOTLIN] Kotlin compilation 'jdkHome' argument: /home/ppalaga/.sdkman/candidates/java/21.0.8-tem
Options for KOTLIN DAEMON: IncrementalCompilationOptions(super=CompilationOptions(compilerMode=INCREMENTAL_COMPILER, targetPlatform=JVM, reportCategories=[0, 3], reportSeverity=2, requestedCompilationResults=[0], kotlinScriptExtensions=[]), sourceChanges=org.jetbrains.kotlin.buildtools.api.SourcesChanges$Known@1fc3580f, classpathChanges=NoChanges, workingDir=/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/kotlin/compileKotlin/cacheable, multiModuleICSettings=MultiModuleICSettings(buildHistoryFile=/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/kotlin/compileKotlin/local-state/build-history.bin, useModuleDetection=false), icFeatures=IncrementalCompilationFeatures(usePreciseJavaTracking=true, withAbiSnapshot=false, preciseCompilationResultsBackup=true, keepIncrementalCompilationCachesInMemory=true, enableUnsafeIncrementalCompilationForMultiplatform=false, enableMonotonousIncrementalCompileSetExpansion=true), outputFiles=[/home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/kotlin/main, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/kotlin/compileKotlin/cacheable, /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/kotlin/compileKotlin/local-state])
Finished executing kotlin compiler using DAEMON strategy
Resolve mutations for :compileJava (Thread[#190,Execution worker Thread 13,5,main]) started.
:compileJava (Thread[#190,Execution worker Thread 13,5,main]) started.
```

Note that lines

```
  Input property 'javaSources' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/Calculator.java has changed.
  Input property 'javaSources' file /home/ppalaga/orgs/cxf/reproducers/i1962-cxf-codegen-issue/build/classes/java/quarkus-generated-sources/wsdl2java/org/tempuri/CalculatorSoap.java has changed.
```

are missing under `compileKotlin` with 3.26.1

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/QuarkusPlugin.java
+++ b/devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/QuarkusPlugin.java
@@ -7,7 +7,6 @@ import java.io.File;
 import java.nio.file.Path;
 import java.util.HashMap;
 import java.util.HashSet;
-import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
@@ -474,7 +473,7 @@ public class QuarkusPlugin implements Plugin<Project> {
                                 // add the code gen sources
                                 final SourceSet generatedSourceSet = sourceSets
                                         .getByName(QuarkusGenerateCode.QUARKUS_GENERATED_SOURCES);
-                                addCodeGenSourceDirs(compileJava, generatedSourceSet, quarkusExt);
+                                addCodeGenSourceDirs(compileJava, generatedSourceSet);
                                 // quarkusGenerateCode is a dependency
                                 compileJava.dependsOn(quarkusGenerateCode);
                                 // quarkusGenerateCodeDev must run before compileJava in case quarkusDev is the target
@@ -495,7 +494,7 @@ public class QuarkusPlugin implements Plugin<Project> {
                                 // add the code gen test sources
                                 final SourceSet generatedSourceSet = sourceSets
                                         .getByName(QuarkusGenerateCode.QUARKUS_TEST_GENERATED_SOURCES);
-                                addCodeGenSourceDirs(compileTestJava, generatedSourceSet, quarkusExt);
+                                addCodeGenSourceDirs(compileTestJava, generatedSourceSet);
                                 compileTestJava.dependsOn(quarkusGenerateCode, quarkusGenerateCodeTests);
                                 if (tasks.contains(new NamedImpl(generatedSourceSet.getCompileJavaTaskName()))) {
                                     compileTestJava.mustRunAfter(tasks.named(generatedSourceSet.getCompileJavaTaskName()));
@@ -511,7 +510,7 @@ public class QuarkusPlugin implements Plugin<Project> {
             tasks.named("compileKotlin", task -> {
                 final SourceSet generatedSourceSet = project.getExtensions().getByType(SourceSetContainer.class)
                         .getByName(QuarkusGenerateCode.QUARKUS_GENERATED_SOURCES);
-                addCodeGenSourceDirs(task, generatedSourceSet, quarkusExt);
+                addCodeGenSourceDirs(task, generatedSourceSet);
                 task.dependsOn(quarkusGenerateCode);
                 task.mustRunAfter(quarkusGenerateCodeDev);
                 if (tasks.contains(new NamedImpl(generatedSourceSet.getCompileJavaTaskName()))) {
@@ -525,7 +524,7 @@ public class QuarkusPlugin implements Plugin<Project> {
             tasks.named("compileTestKotlin", task -> {
                 final SourceSet generatedSourceSet = project.getExtensions().getByType(SourceSetContainer.class)
                         .getByName(QuarkusGenerateCode.QUARKUS_TEST_GENERATED_SOURCES);
-                addCodeGenSourceDirs(task, generatedSourceSet, quarkusExt);
+                addCodeGenSourceDirs(task, generatedSourceSet);
                 task.dependsOn(quarkusGenerateCodeTests);
                 if (tasks.contains(new NamedImpl(generatedSourceSet.getCompileJavaTaskName()))) {
                     task.mustRunAfter(tasks.named(generatedSourceSet.getCompileJavaTaskName()));
@@ -538,22 +537,14 @@ public class QuarkusPlugin implements Plugin<Project> {
         });
     }
 
-    private static void addCodeGenSourceDirs(JavaCompile compileJava, SourceSet generatedSourceSet,
-            QuarkusPluginExtension quarkusExt) {
+    private static void addCodeGenSourceDirs(JavaCompile compileJava, SourceSet generatedSourceSet) {
         final File baseDir = generatedSourceSet.getJava().getClassesDirectory().get().getAsFile();
-        for (String provider : quarkusExt.getCodeGenerationProviders().get()) {
-            compileJava.source(new File(baseDir, provider));
-        }
+        compileJava.source(baseDir);
     }
 
-    private static void addCodeGenSourceDirs(Task compileKotlin, SourceSet generatedSourceSet,
-            QuarkusPluginExtension quarkusExt) {
+    private static void addCodeGenSourceDirs(Task compileKotlin, SourceSet generatedSourceSet) {
         final File baseDir = generatedSourceSet.getJava().getClassesDirectory().get().getAsFile();
-        final List<String> codeGenProviders = quarkusExt.getCodeGenerationProviders().get();
-        final Object[] codeGenDirs = new Object[codeGenProviders.size()];
-        for (int i = 0; i < codeGenDirs.length; ++i) {
-            codeGenDirs[i] = new File(baseDir, codeGenProviders.get(i));
-        }
+        final Object[] codeGenDirs = new Object[] { baseDir };
         try {
             var sourcesMethod = compileKotlin.getClass().getMethod("source", Object[].class);
             sourcesMethod.invoke(compileKotlin, new Object[] { codeGenDirs });

